### PR TITLE
Feature/ghela alert blogpost

### PR DESCRIPTION
**Pull request by basmoorkens**
### Thank you for your contribution! Please go through this small checklist to prepare your pull request to be reviewed as soon as possible :)
* [x] All authors were added to `_data/authors.yml` with a picture and an up-to-date bio
* [x] The blog post image is of ratio 500x293
  * Feel free to leave a comment if you don't know how to resize your image and someone will gladly help you out
* [x] The [**one sentence per line** method](https://raw.githubusercontent.com/brandon-rhodes/blog/master/texts/brandon/2012/one-sentence-per-line.rst) has been adopted
* [x] A spell check has been done
* [x] All links open in a new tab
  * Example: `[Amazing tech blog](https://ordina-jworks.github.io){:target="_blank" rel="noopener noreferrer"}`
* [x] The post's content properly resizes according to the window's size
  * Tip: Just shorten your window's horizontal length to easily test this
  * To have images properly resize use `class="image fit"`
  * To have YouTube videos properly resize, wrap the iframe into a `<div class="responsive-video">`
* [x] If a table of content is: check whether all links are working correctly

### Any specific remarks for the reviewers or anything you need help with?
\<enter text here>

### Checklist for the reviewers before merging
* [x] Publishing date of the post has been correctly set
  * This can be done by renaming the filename or by setting the date in the header section of the post

### CODE REVIEW

File: _posts/2020-09-30-ghelamco-alert.md [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/h2-console.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+#### Metrics
+{:.no_toc}
+
+Our application generates metrics to provide us with some data about our backend.  
+We collect data and push it into AWS cloudwatch as metrics for:
+* Number of events added
+* Number of events update
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:144]:
```suggestion
* Number of events updated
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+* Number of events cancelled
+* Number of alerts snoozed
+* Number of alerts started
+* Number of alerts stopped
+* A basic heartbeat
+
+When our application is running, we are constantly gathering this data which we push to Cloudwatch metrics, every 5 minutes.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:151]:
```suggestion
When our application is running, we are constantly gathering this data which we push to Cloudwatch metrics every 5 minutes.
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 3]
```diff
@@ -0,0 +1,824 @@
...
+Afterwards we reset our data counters, and start collecting again.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:152]:
```suggestion
Afterwards we reset our data counters and start collecting again.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 4]
```diff
@@ -0,0 +1,824 @@
...
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/h2-console.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+#### Metrics
+{:.no_toc}
+
+Our application generates metrics to provide us with some data about our backend.  
+We collect data and push it into AWS cloudwatch as metrics for:
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:142]: Add a link to the aws cloudWatch page

File: _posts/2020-09-30-ghelamco-alert.md [hunk 5]
```diff
@@ -0,0 +1,824 @@
...
+We use these credentials inside our application to access the AWS SDK from our java code.  
+We integrate with cloudwatch in two different ways:
+* We push custom metrics from our application to cloudwatch metrics.
+* We send the logs from our application to cloudwatch logs.
+
+##### Cloudwatch metrics
+{:.no_toc}
+
+The metrics that we generate in our application should of course get sent to AWS cloudwatch, so we can actually make use of them in the AWS cloud.  
+We used the metric data to build dashboards and alerts within cloudwatch for added visibility into our application.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:173]:
```suggestion
We use the metric data to build dashboards and alerts within cloudwatch for added visibility into our application.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 6]
```diff
@@ -0,0 +1,824 @@
...
+This allows us to check the log files in the AWS cloud without needing access to the RPI device itself.  
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/cloudwatch-logs.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+
+#### AWS IoT
+{:.no_toc}
+
+The reason for using AWS IoT is for the sake of "being connected to the cloud".
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:200]: add a link to aws iot

File: _posts/2020-09-30-ghelamco-alert.md [hunk 7]
```diff
@@ -0,0 +1,824 @@
...
+Our thing-certificate and our private key are our credentials when we try to communicate with AWS IoT to access our edge device.  
+The only additional input that we need to provide on top of the generated signing request is the root certificate to check the signing.  
+
+* Certificates
+* private Key
+* Root CA
+
+Since these certificates get generated per thing that you register they should only be used for 1 device.  
+When I developed the application I had to register another thing for my laptop.  
+Connecting from the RPI and the laptop with the same certificates caused some unwanted behavior like connection interrupts.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:226]: Can this be detected in aws iot?

Would you be able to take over a connection by having another device send in data?

Just some food for thought

File: _posts/2020-09-30-ghelamco-alert.md [hunk 8]
```diff
@@ -0,0 +1,824 @@
...
+When we are connected to the cloud, we are able to communicate with our device "through the cloud", meaning from anywhere we want!   
+In our case this means that we want our RPI device to be connected to AWS IoT whenever it is up and running and has internet connectivity.  
+Of course the communication between our RPI and the AWS cloud has to be secured.
+AWS IoT uses authentication and authorization workflow by using certificates that you issue from AWS IoT and upload to your edge device.  
+We treat our RPI as an edge device in this project.
+
+Authentication works based on the certificates that the edge device presents to AWS IoT.  
+Once the certificate authentication is successful, AWS IoT checks which policies got attached to those certificates to grant it specific authorizations within the AWS cloud.  
+
+##### Authentication: certificates
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:210]: lots of things in this section

File: _posts/2020-09-30-ghelamco-alert.md [hunk 9]
```diff
@@ -0,0 +1,824 @@
...
+Connecting from the RPI and the laptop with the same certificates caused some unwanted behavior like connection interrupts.  
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/cert-aws-iot.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+##### Authorization: policies
+{:.no_toc}
+
+After creating our certificates and keys, we need to handle the authorization part.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:235]:
```suggestion
After creating our certificates and keys we need to handle the authorization part.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 10]
```diff
@@ -0,0 +1,824 @@
...
+- "iot:UpdateThingShadow"
+- "iot:DeleteThingShadow" 
+
+This system of generating certificates and coupling policies is a very secure and easy way of working with edge devices.   
+
+##### jobs
+{:.no_toc}
+
+A crucial part of AWS IoT is the job section. 
+We can create a jobs by using the AWS cli.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:293]: a job or jobs?

File: _posts/2020-09-30-ghelamco-alert.md [hunk 11]
```diff
@@ -0,0 +1,824 @@
...
+  --presigned-url-config {roleArn:arn:aws:iam::123456789012:role/S3DownloadRole, expiresInSec:3600}
+```
+We use these jobs AWS Iot to create these jobs and send them to our RPI where they get executed by our application code.  
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert device update" src="/img/2020-09-25-ghelamco-alert/device-update-iot.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+We have the possibility to put multiple "things" in a group of devices.  
+This way we can send jobs to the entire group and the job gets executed on every device from that group.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:310]:
```suggestion
This way we can send jobs to the entire group and these jobs get executed on every device from that group.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 12]
```diff
@@ -0,0 +1,824 @@
...
+    "operation": "snoozeAlert",
+    "local_alert_id": 31,
+    "event_title": "test from lambda snooze alert"
+}
+```
+
+##### MQTT protocol
+{:.no_toc}
+
+AWS IoT can communicate with it's registered devices through 2 protocols: HTTP or MQTT.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:341]:
```suggestion
AWS IoT can communicate with its registered devices through 2 protocols: HTTP or MQTT.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 13]
```diff
@@ -0,0 +1,824 @@
...
+The broker is a server that receives all messages from the clients and then routes those messages to relevant destination clients.  
+A client is anything that can interact with the broker to send and receive messages.  
+Our client is the RPI but it could also be an IoT sensor in the field or an application in a data center that processes IoT data.  
+
+In our backend java code we have an MQTTJobService which connects to AWS IoT by using the [AWS SDK](https://aws.amazon.com/sdk-for-java/) and subscribes to the relevant topics to receive and respond to AWS Iot jobs.  
+Every 30 seconds we will read these topics to see if there are any new jobs to be processed.  
+
+### AWS IOT Greengrass
+
+AWS greengrass is a service that extends the AWS cloud onto your edge device.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:366]: add a link to greengrass

File: _posts/2020-09-30-ghelamco-alert.md [hunk 14]
```diff
@@ -0,0 +1,824 @@
...
+In our backend java code we have an MQTTJobService which connects to AWS IoT by using the [AWS SDK](https://aws.amazon.com/sdk-for-java/) and subscribes to the relevant topics to receive and respond to AWS Iot jobs.  
+Every 30 seconds we will read these topics to see if there are any new jobs to be processed.  
+
+### AWS IOT Greengrass
+
+AWS greengrass is a service that extends the AWS cloud onto your edge device.  
+This was fascinating for us since we had some tough problems to solve:  
+* How can we deploy our application on the RPI device?  
+* How can we make sure our application recovers from failures?
+* How can we keep our system itself up to date?
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:370]:
```suggestion
* How can we make sure that our system keeps itself up to date?
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 15]
```diff
@@ -0,0 +1,824 @@
...
+
+The greengrass group allows us to push deployments to all the devices in our group.  
+It also allows you to run lambda functions on your core device or install additional connectors with your own runtime of choice.  
+<div style="text-align: center;">
+  <img alt="Greengrass group" src="/img/2020-09-25-ghelamco-alert/gg_group.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+When you register your device as a core device in greengrass you immediately get some nice additional benefits from this.  
+For example, you can immediately see all the network interfaces on your core device and what ip addresses got allocated.  
+This is especially useful if you want to ssh to your device and do not have fixed ip attached to it.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:387]:
```suggestion
This is especially useful if you want to ssh to your device and do not have a fixed ip attached to it.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 16]
```diff
@@ -0,0 +1,824 @@
...
+</div>
+
+When you register your device as a core device in greengrass you immediately get some nice additional benefits from this.  
+For example, you can immediately see all the network interfaces on your core device and what ip addresses got allocated.  
+This is especially useful if you want to ssh to your device and do not have fixed ip attached to it.  
+<div style="text-align: center;">
+  <img alt="Greengrass core" src="/img/2020-09-25-ghelamco-alert/gg_core.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+Now we that greengrass is installed and our RPI is configured as a core device our greengrass group we can start making full use of the capabilities that greengrass offers.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:392]:
```suggestion
Now that greengrass is installed and our RPI is configured as a core device on our greengrass group, we can start making full use of the capabilities that greengrass offers.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 17]
```diff
@@ -0,0 +1,824 @@
...
+
+When you register your device as a core device in greengrass you immediately get some nice additional benefits from this.  
+For example, you can immediately see all the network interfaces on your core device and what ip addresses got allocated.  
+This is especially useful if you want to ssh to your device and do not have fixed ip attached to it.  
+<div style="text-align: center;">
+  <img alt="Greengrass core" src="/img/2020-09-25-ghelamco-alert/gg_core.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+Now we that greengrass is installed and our RPI is configured as a core device our greengrass group we can start making full use of the capabilities that greengrass offers.  
+We decided that we wanted our application to run as a docker container, so we installed and configured the **docker connector** for greengrass.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:393]: Add a link to that docker connector page

like: https://docs.aws.amazon.com/greengrass/latest/developerguide/docker-app-connector.html

File: _posts/2020-09-30-ghelamco-alert.md [hunk 18]
```diff
@@ -0,0 +1,824 @@
...
+</div>
+For this to work you have to install docker and docker compose on your core device.  
+Since we are using a RPI it was a bit harder to install since the RPI uses the ARM7 chipset instead of x64/x32.  
+We had no trouble installing docker itself, but for docker compose we had to clone the source code from github and compile the binaries ourself on the RPI.  
+After all the setup was done we could just create a docker container in our CICD pipeline and tell greengrass to run that container on RPI.  
+
+### CICD pipeline RPI Backend
+
+To make sure we did not need to bother ourselves with manual builds and installs of our code on the RPI we built a CICD pipeline to automatically deploy our software onto the RPI.  
+We trigger the pipeline whenever a push to our master branch in our git repository.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:406]:
```suggestion
We trigger the pipeline whenever a commit is pushed to our master branch in our git repository.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 19]
```diff
@@ -0,0 +1,824 @@
...
+For this to work you have to install docker and docker compose on your core device.  
+Since we are using a RPI it was a bit harder to install since the RPI uses the ARM7 chipset instead of x64/x32.  
+We had no trouble installing docker itself, but for docker compose we had to clone the source code from github and compile the binaries ourself on the RPI.  
+After all the setup was done we could just create a docker container in our CICD pipeline and tell greengrass to run that container on RPI.  
+
+### CICD pipeline RPI Backend
+
+To make sure we did not need to bother ourselves with manual builds and installs of our code on the RPI we built a CICD pipeline to automatically deploy our software onto the RPI.  
+We trigger the pipeline whenever a push to our master branch in our git repository.  
+We used azure devops as our CICD system.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:407]: link to azure devops

Note: why azure devops and doesn't aws have something similar?

File: _posts/2020-09-30-ghelamco-alert.md [hunk 20]
```diff
@@ -0,0 +1,824 @@
...
+As you can see there are 2 major parts of this pipeline: 
+* the azure devops pipeline
+* the greengrass deployment process.  
+
+#### Azure devops pipeline
+{:.no_toc}
+
+* Build and test our java code.  
+We use maven to build and test our code.   
+* We use [AWS SSM - parameter store]() to safely store our config files and certificates for the RPI.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:428]: missing link to aws ssm?

File: _posts/2020-09-30-ghelamco-alert.md [hunk 21]
```diff
@@ -0,0 +1,824 @@
...
+* the greengrass deployment process.  
+
+#### Azure devops pipeline
+{:.no_toc}
+
+* Build and test our java code.  
+We use maven to build and test our code.   
+* We use [AWS SSM - parameter store]() to safely store our config files and certificates for the RPI.  
+In step 2 and 3 we fetch those config files and secrets from the parameter store and store them locally for use later in the pipeline.  
+* In order to control the alert light hooked up to our RPI we need to download the wiringPi library as we used that to communicate with our alert light.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:430]: add a link to the wiringPi library

File: _posts/2020-09-30-ghelamco-alert.md [hunk 22]
```diff
@@ -0,0 +1,824 @@
...
+
+#### Azure devops pipeline
+{:.no_toc}
+
+* Build and test our java code.  
+We use maven to build and test our code.   
+* We use [AWS SSM - parameter store]() to safely store our config files and certificates for the RPI.  
+In step 2 and 3 we fetch those config files and secrets from the parameter store and store them locally for use later in the pipeline.  
+* In order to control the alert light hooked up to our RPI we need to download the wiringPi library as we used that to communicate with our alert light. 
+We stored this library in our releases S3 bucket.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:431]: releases or release?

File: _posts/2020-09-30-ghelamco-alert.md [hunk 23]
```diff
@@ -0,0 +1,824 @@
...
+* We expose our application port 8080 to the outside world.
+* We mount our AWS credentials and H2 database as volumes from the RPI host system.
+
+#### Greengrass deployment process
+{:.no_toc}
+
+For the greengrass deployment process we did not have to create any code.  
+This works out of the box when you set up greengrass.  
+However, for full transparency I will briefly describe the process here:
+* AWS greengrass receives a new deployment ( this is the last step of our azure devops pipeline).
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:489]:
```suggestion
* AWS greengrass receives a new deployment (this is the last step of our azure devops pipeline).
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 24]
```diff
@@ -0,0 +1,824 @@
...
+First let's take a look into our frontend application.  
+
+### Angular
+I wanted to create a simple web application to get more insight in the data of our RPI device.  
+Any modern frontend framework would be suitable for this but I decided to go with Angular since I had already had some basic experience with it.  
+We did however use some extra frameworks and AWS services to make it easier to build our frontend application.  
+
+### AWS Amplify
+
+To help us bootstrap our web application, we decided to use the AWS amplify framework.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:516]: add a link to aws amplify

File: _posts/2020-09-30-ghelamco-alert.md [hunk 25]
```diff
@@ -0,0 +1,824 @@
...
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert jobs screen" src="/img/2020-09-25-ghelamco-alert/frontend-jobs.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+#### Metrics
+{:.no_toc}
+
+On this tab we can see the status of the RPI backend as a graph.  
+Remember the Metrics we talked about before?
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:615]:
```suggestion
Remember the metrics we talked about before?  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 26]
```diff
@@ -0,0 +1,824 @@
...
+We will explain the different parts of our serverless setup to show you in detail how everything works. 
+
+#### Functions and Api Gateway
+{:.no_toc}
+
+The essential part of our serverless setup is the functions section.  
+A function is an AWS Lambda function.  
+It's an independent unit of deployment like a true microservice.  
+It is generally a very small piece of code that does one thing and does it well.  
+In our project for example we have a lambda function **list-events** which does exactly that, list the events from our dynamoDB table.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:699]:
```suggestion
In our project, for example, we have a lambda function **list-events** which does exactly that, list the events from our dynamoDB table.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 27]
```diff
@@ -0,0 +1,824 @@
...
+5. The job gets send to the RPI and processed by our backend application, thus snoozing the alert in the **H2 database**.
+This will effectively turn the alert light off.    
+6. The RPI **synchronizes** his H2 database with the event **dynamoDB table**. Thus updating the status of the alert in the dynamoDB table.  
+7. This new status will then be reflected in the **frontend web application** so our user can see the result of his action.  
+
+This same workflow applies to all commands that can be sent from the frontend application to our RPI device.  
+This is a prime example of how a distributed system works and operates.  
+
+# Conclusion
+This internship gave me such a big cover of all best practices and new technologies, that this was a real eye-opening and educational experience.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:801]:
```suggestion
This internship gave me such a big overview of all best practices and new technologies, that this was a real eye-opening and educational experience.  
```

Or: This internship covered such a large amount of best practices and new technologies that this was a real eye-opening educational experience.

File: _posts/2020-09-30-ghelamco-alert.md [hunk 28]
```diff
@@ -0,0 +1,824 @@
...
+- API's, connecting it all together...
+- came in touch with a lot of linux, sharpening my commandline skills
+- developed an Angular app from start to finish
+- learned a lot about AWS: IAM, IoT Core, DynamoDB, API Gateway, SNS, SSM, EKS, S3, Cloudwatch, Cloudfront, lambdas, Route53, ...
+- CICD pipelines and their best practices in the industry
+
+I'm very thankful for being able to do this and having the chance to work with Bas on a real project.  
+
+He always motivated me to go the extra mile !  
+He could destroy a day of hard work in a few sentences... :)
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:814]: dat is hoe dat we hem kennen 😉

Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:814]:
```suggestion
And he could destroy a day of hard work in a few sentences ... :)  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 29]
```diff
@@ -0,0 +1,824 @@
...
+I'm very thankful for being able to do this and having the chance to work with Bas on a real project.  
+
+He always motivated me to go the extra mile !  
+He could destroy a day of hard work in a few sentences... :)  
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/bas-smile.png" width="auto" height="auto" target="_blank" >
+</div>
+
+But afterwards, he would always take the time to explain why, and how I should do it to make my solution comply with the industries best practices.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:820]:
```suggestion
But afterwards he would always take the time to explain why, and how, I should do it to make my solution comply with the industry best practices.  
```

File: _posts/2020-09-30-ghelamco-alert.md [hunk 30]
```diff
@@ -0,0 +1,824 @@
...
+
+He always motivated me to go the extra mile !  
+He could destroy a day of hard work in a few sentences... :)  
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/bas-smile.png" width="auto" height="auto" target="_blank" >
+</div>
+
+But afterwards, he would always take the time to explain why, and how I should do it to make my solution comply with the industries best practices.  
+Being able to have someone who makes you self-reflect on the work you did, and takes his time to give you proper feedback, is the biggest asset to provide to a junior programmer.
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:821]:
```suggestion
Being able to have someone who makes you self-reflect on the work you did and takes his time to give you proper feedback.
That is the biggest asset to provide to a junior programmer.  
```

**Review by tomvdbulck — COMMENTED** [review]
Looking good, some small comments, maybe add some extra links - some of these seem to show up later, but some extra links doesn't hurt.

File: _posts/2020-09-30-ghelamco-alert.md [hunk 31]
```diff
@@ -0,0 +1,824 @@
+---
+layout: post
+authors: [kevin_govaerts, bas_moorkens]
+title: 'Ghelamco alert: '
+image: /img/2020-09-25-ghelamco-alert/rpi-front.jpg
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:5]: this does not have the correct size: 500x293

File: _posts/2020-09-30-ghelamco-alert.md [hunk 32]
```diff
@@ -0,0 +1,824 @@
...
+The Raspberry PI will run an application that parses the game fixtues on the website of KAA Gent to search for any home fixtures.  
+Whenever a home fixture occurs, the RPI will start turning on the alert light on a preset schedule.  
+This way the employees in the office will have a visual warning on game day and they will be reminded to leave the parking lot on time.  
+Since we wanted to build a user friendly solution we added a serverless backend in the AWS cloud combined with an angular application so that our users can also look at the web application to get some additional feedback and even manipulate the alerting schedule.  
+This enables users to do some additional operations through the web application:
+* Snooze alerts
+* Create additional alerts
+* Create custom events
+
+<!-- //TODO GIF Proof of working project -->
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:39]: TODO?

File: _posts/2020-09-30-ghelamco-alert.md [hunk 33]
```diff
@@ -0,0 +1,824 @@
...
+- learned a lot about AWS: IAM, IoT Core, DynamoDB, API Gateway, SNS, SSM, EKS, S3, Cloudwatch, Cloudfront, lambdas, Route53, ...
+- CICD pipelines and their best practices in the industry
+
+I'm very thankful for being able to do this and having the chance to work with Bas on a real project.  
+
+He always motivated me to go the extra mile !  
+He could destroy a day of hard work in a few sentences... :)  
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/bas-smile.png" width="auto" height="auto" target="_blank" >
```
Comment (tomvdbulck) [inline _posts/2020-09-30-ghelamco-alert.md:817]: image not perfectly cut out, maybe put it on a white background to remove the pixelated effect at the borders?

**Review by tomvdbulck — COMMENTED** [review]
Looking good - only some comments on some images

File: _posts/2020-09-30-ghelamco-alert.md (hunk 28 above)
Comment (basmoorkens) [reply _posts/2020-09-30-ghelamco-alert.md]: haha :)

File: _posts/2020-09-30-ghelamco-alert.md (hunk 7 above)
Comment (basmoorkens) [reply _posts/2020-09-30-ghelamco-alert.md:224]: yeah, if you posses the certificates you have the keys to the connection. best practice is probably not to share certificates but it was easier this way in our dev phase

File: _posts/2020-09-30-ghelamco-alert.md (hunk 19 above)
Comment (basmoorkens) [reply _posts/2020-09-30-ghelamco-alert.md:405]: because bousson has an azure fetish

**Review by yannickdeturck — APPROVED** [review]
Good stuff, @kevingovaerts and @basmoorkens !
Very interesting use case and architecture. 👌🏼

Did some polishing. Post looks good to go. :)

**Pull request merged**